src/main/java/net/minecraftforge/common/BiomeManager.java

```diff
@@ -123,7 +139,22 @@ public static void removeBiome(BiomeType type, BiomeEntry entry)
 
         return list != null ? ImmutableList.copyOf(list) : null;
     }
+*/
+    public static ImmutableList<Pair<Biome.Attributes, Supplier<Biome>>> getNetherBiomes(Registry<Biome> biomeRegistry, Pair<Biome.Attributes, Supplier<Biome>>... biomes)
+    {
+        ImmutableList.Builder<Pair<Biome.Attributes, Supplier<Biome>>> result = ImmutableList.builder();
+        result.add(biomes);
+
+        synchronized (moddedNetherBiomes)
+        {
+            for (Pair<RegistryKey<Biome>, Biome.Attributes> moddedBiome : moddedNetherBiomes) {
+                result.add(Pair.of(moddedBiome.getSecond(), () -> biomeRegistry.func_243576_d(moddedBiome.getFirst())));
```

Comment:
You can do moddedBiome.swap().mapSecond(s -> ), a bit cleaner imo.

Reply:
A bit more relevant `func_243576_d` is `getOrThrow`, which is fine for vanilla biomes but will crash on datapack biomes.
This is because there are multiple registries that exist for the same type, vanilla only are the static ones in `WorldGenRegistries`, while datapack ones are held in `DatapackRegistries` and are rebuilt.
The way it works is that pure vanilla is loaded once, after the first "Create new world" button click, and a copy of the vanilla only registry will be passed here (through `DatapackRegistries`, making datapack biomes crash since they are not present.
When clicking the other "Create new world" button, the objects are built again but this time including the objects coming from datapacks.
Basically, datapack biomes should get loaded, but won't because it'll crash the first time.
You could simply use one of the function that returns an optional and check if present first, but it's a bit dirty.

Reply:
So, in order to support datapack biomes, should I use a different method? I'm trying to make it work so that datapack biomes can be added with this.

Reply:
I mean, it probably depends on when the supplier is resolved

Reply:
I mean you can test it https://gist.github.com/Cyborgmas/fa514d3297125e98ffb8377865891c2b and see what happens and find a proper fix. Otherwise I am working on another system that'll allow to modify it in a more data-driven way.

Reply:
If you're making modded nether biomes data driven there's likely be no further need for this pr since forge afaik has said they want to head that way I think. Though I will maintain this PR until another PR opens.